appinventor/components/src/com/google/appinventor/components/runtime/Regression.java

```diff
@@ -0,0 +1,125 @@
+// -*- mode: java; c-basic-offset: 2; -*-
+// Copyright 2022-2023 MIT, All rights reserved
+// Released under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package com.google.appinventor.components.runtime;
+
+import com.google.appinventor.components.annotations.*;
+import com.google.appinventor.components.common.*;
+import com.google.appinventor.components.runtime.util.YailDictionary;
+import com.google.appinventor.components.runtime.util.YailList;
+import gnu.lists.LList;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * A data science component to apply different regression models
+ * The component requires a data source to apply the model on
+ */
+@DesignerComponent(version = YaVersion.REGRESSION_COMPONENT_VERSION,
+    description = "A component that contains regression models",
+    category = ComponentCategory.DATASCIENCE,
+    iconName = "images/web.png",
+    nonVisible = true)
+@SimpleObject
+@SuppressWarnings("checkstyle:JavadocParagraph")
+public final class Regression extends DataCollection {
+
+  /**
+   * Creates a new Regression component.
+   */
+  public Regression(ComponentContainer container) {
+    super();
+    // Construct default dataFileColumns list with 2 entries
+    dataFileColumns = Arrays.asList("", "");
+    sheetsColumns = Arrays.asList("", "");
+    webColumns = Arrays.asList("", ""); // Construct default webColumns list with 2 entries
```

Comment:
See previous comments on AnomalyDetection and ChartData2D about how these fields are initialized.